src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -168,17 +192,37 @@ public String getNamespace(Interface apiInterface) {
 
   /** The function name to set the given proto field. */
   public String getFieldSetFunctionName(FeatureConfig featureConfig, FieldConfig fieldConfig) {
-    Field field = fieldConfig.getField();
-    if (featureConfig.useResourceNameFormatOption(fieldConfig)) {
-      return getResourceNameFieldSetFunctionName(fieldConfig.getMessageFieldConfig());
-    } else {
-      return getFieldSetFunctionName(field);
+    FieldType field = fieldConfig.getField();
+    switch (fieldConfig.getField().getApiSource()) {
+      case PROTO:
+        if (featureConfig.useResourceNameFormatOption(fieldConfig)) {
+          return getResourceNameFieldSetFunctionName(fieldConfig.getMessageFieldConfig());
+        } else {
+          return getFieldSetFunctionName(field);
+        }
+      case DISCOVERY:
+        if (featureConfig.useResourceNameFormatOption(fieldConfig)) {
+          return getResourceNameFieldSetFunctionName(fieldConfig.getMessageFieldConfig());
+        } else {
+          return getFieldSetFunctionName(field);
+        }
+      default:
+        throw new IllegalArgumentException("Unhandled model type.");
     }
   }
 
   /** The function name to set the given proto field. */
-  public String getFieldSetFunctionName(Field field) {
-    return getFieldSetFunctionName(field.getType(), Name.from(field.getSimpleName()));
+  public String getFieldSetFunctionName(FieldType field) {
+    return getFieldSetFunctionName(field.getProtoTypeRef(), Name.from(field.getSimpleName()));
+  }
+
+  /** The function name to set a field from a Discovery document. */
+  public String getFieldSetFunctionName(Schema schema) {
```

Comment:
It's a little more legwork than I'd like to remove the TypeRef function, but I do want to keep the FieldType function. 

I've just removed the Schema function, put a TODO to remove the TypeRef function later, and changed the implementation of the FieldType function so that it doesn't call field.getProtoTypeRef() because yes that is gross.